content/asset-tracking/using-the-sdks.textile

```diff
@@ -317,6 +405,13 @@ subscriber.trackableStates
     .launchIn(scope) // coroutines scope on which the statuses are received
 ```
 
+```[swift]
+// Really not sure about this one!
+let event as AddTrackableEvent: self?.performAddTrackableEvent(event){
```

Comment:
To listen to events, you need to implement some or all of the delegate methods declared in `SubscriberDelegate`. You can either create a separate class (e.g. the example below, I define `SubscriberDelegate`) or you can make your current class (e.g. AppDelegate) conform to `SubscriberDelegate` add stick the delegate methods in the same class followed by `subscriber.delegate = self`. iOS developers would probably use implement the delegate methods in an [extension](https://docs.swift.org/swift-book/LanguageGuide/Extensions.html).

My example creates a dedicated class to implement the delegate methods:

```swift
import AblyAssetTrackingSubscriber

class MySubscriberDelegate: SubscriberDelegate {
    // Implement some or all of the delegate methods
    override func subscriber(sender: Subscriber, didChangeAssetConnectionStatus status: ConnectionState) {
        /* Handle the change */
    }
    
    override func subscriber(sender: Subscriber, didFailWithError error: ErrorInformation) {
        /* Handle the error */
    }
    
    override func subscriber(sender: Subscriber, didUpdateEnhancedLocation location: CLLocation) {
        /* Handle the location update */
    }
}
```

Somewhere else:
```swift
subscriber.delegate = MySubscriberDelegate();
```

You also don't need to set the field yourself. You can pass it in the builder (`SubscriberBuilder` has a function called `delegate()`. You had this in another snippet:
> `.delegate(self) // provide a delegate to handle received location updates`

Reply:
A huge help. Thanks @ben-xD